utils/python/CIME/XML/env_batch.py

```diff
@@ -67,6 +69,34 @@ def get_type_info(self, vid):
     def get_jobs(self):
         result = []
         for node in self.get_nodes("job"):
-            result.append(node.get("name"))
-
+            name = node.get("name")
+            template = self.get_value("template", subgroup=name)
+            task_count = self.get_value("task_count", subgroup=name)
+            result.append((name, template, task_count))
         return result
+
+    def create_job_groups(self, bjobs):
+        # only the job_submission group is repeated
+        group = self.get_node("group", {"id":"job_submission"})
+        # look to see if any jobs are already defined
+        cjobs = self.get_jobs()
+        childnodes = list()
+
+        expect(len(cjobs)==0," Looks like job groups have already been created")
+
+        for child in reversed(group):
+            childnodes.append(deepcopy(child))
```

Comment:
The function is different.   Here we are repeating the nodes in the job_submission group once for each job listed in config_batch.xml